### PR TITLE
Add license information to JS files

### PR DESCRIPTION
**Pull request by Robin-Wils**
## Why you might want this
GNU LibreJS notices licenses like this.  
The JS does currently not work for people who use the GNU LibreJS add-on.  
This commit should fix that.

## Why is the request directly to the master branch?
I can recreate this request if you prefer to have it on the dev branch first.
I decided to create the request to master since it only adds a few comments.

### CODE REVIEW

File: js/underscore-min.js [hunk 1]
```diff
@@ -2,5 +2,7 @@
...
-//# sourceMappingURL=underscore-min.map
\ No newline at end of file
+// @license-end
+//# sourceMappingURL=underscore-min.map
```
Comment (RDIL) [inline js/underscore-min.js]: This file is minified on purpose, please remove the extra blank line at the end.  Other then that, good change! 👍

**Comment by MrWils** [comment]
I remove that white line.
(Just in case that you didn't notice)

**Comment by jancborchardt** [comment]
The modifications on the Bootstrap and Underscore files will be overridden on update of the library, and I’ m not really a fan of modifying library release files. That’s something which should happen upstream at the libraries instead?

**Comment by MrWils** [comment]
There exists another solution which might solve that problem.  
https://www.gnu.org/software/librejs/free-your-javascript.html#step3

It might be better to do that on a dev branch first if this is a wanted "feature".

**Comment by MrWils** [comment]
There exists another solution which might solve that problem.  
https://www.gnu.org/software/librejs/free-your-javascript.html#step3

It might be better to do that on a dev branch first if this is a wanted "feature".

**Comment by jancborchardt** [comment]
This approach seems quite unscalable. Why should every single website mark the license of well known open source libraries like Bootstrap and Underscore? LibreJS should rather have an automated allowlist of those.

Sorry, but it seems like this just creates unnecessary busywork on part of the maintainers of many open source projects. Or what am I missing here?

**Comment by MrWils** [comment]
Well known scripts like the most bootstrap scripts are accepted by default.   
Underscore isn't accepted by default though and "site.js" isn't either.  

Besides that, JavaScript can be under another license as the project.    
The icecat browser has LibreJS enabled by default, some repo's don't even have other browsers. Those users won't be able to run the JS.  
  
Setting the JS free is a choice but it is ok to close this request if this is too much work for the maintainers. It is only one small thing extra to maintain but I understand that a few small things make a big one.

**Comment by Jc-L** [comment]
@jancborchardt Should we close this or investigate further ?

**Comment by RDIL** [comment]
I'm not sure what to say on this PR. Added `maybe` label.

**Comment by Jc-L** [comment]
As mentionned by @jancborchardt, it puts workload on the site maintainers. Value is debatable if we consider the impacted audience, even if the goal is understandable. I'd rather put the topic on LibreJS to include common libraries, like `underscore`, to begin with.
My opinion is to eject the request, but I am not a reviewer anyway :)

**Comment by RDIL** [comment]
@Jc-L if you want to join the review team, here:
https://orgmanager.miguelpiedrafita.com/o/orgmanager

**Comment by RDIL** [comment]
sorry wrong link, edited it

**Comment by jancborchardt** [comment]
Yeah, for reasons mentioned above I'll close the request.

LibreJS should accept common free & open source libraries, as well as check "LICENSE" files of websites. Otherwise this is neither scalable nor maintainable.

**Pull request closed without merging**